web_notify/models/res_users.py

```diff
@@ -23,24 +25,35 @@ def _compute_channel_names(self):
         compute='_compute_channel_names')
 
     @api.multi
-    def notify_info(self, message, title=None, sticky=False):
+    def notify_info(self, message, title=None, sticky=False, commit=False):
         title = title or _('Information')
         self._notify_channel(
-            'notify_info_channel_name', message, title, sticky)
+            'notify_info_channel_name', message, title, sticky, commit=commit)
 
     @api.multi
-    def notify_warning(self, message, title=None, sticky=False):
+    def notify_warning(self, message, title=None, sticky=False, commit=False):
         title = title or _('Warning')
         self._notify_channel(
-            'notify_warning_channel_name', message, title, sticky)
+            'notify_warning_channel_name', message, title, sticky,
+            commit=commit)
 
     @api.multi
-    def _notify_channel(self, channel_name_field, message, title, sticky):
+    def _notify_channel(self, channel_name_field, message, title, sticky,
+                        commit=False):
         bus_message = {
             'message': message,
             'title': title,
             'sticky': sticky
         }
         notifications = [(getattr(record, channel_name_field), bus_message)
                          for record in self]
-        self.env['bus.bus'].sendmany(notifications)
+        if commit:
+            with odoo.api.Environment.manage():
+                with odoo.registry(self.env.cr.dbname).cursor() as new_cr:
+                    new_env = api.Environment(new_cr, self.env.uid,
+                                              self.env.context)
+                    self.env['bus.bus'].with_env(new_env).sendmany(
+                        notifications)
+                    new_env.cr.commit()
```

Comment:
Hi @Yajo 

Thanks for your comment.

> This is implicit, no need for it.

I wrote this code like the exemple in the last paragraph of the section in the guidelines :

> You can isolate a transaction for a valid cr.commit using Environment:
```
with openerp.api.Environment.manage():
      with openerp.registry(self.env.cr.dbname).cursor() as new_cr:
          # Create a new environment with new cursor database
          new_env = api.Environment(new_cr, self.env.uid, self.env.context)
          # with_env replace original env for this method
          self.with_env(new_env).write({'name': 'hello'})  # isolated transaction to commit
          new_env.cr.commit()  # Don't show a invalid-commit in this case
      # You don't need close your cr because is closed when finish "with"
  # You don't need clear caches because is cleared when finish "with"
```